# Andox individual project planning

In [17]:
library(tidyverse)
library(repr)
library(tidymodels)
library(cowplot)
options(repr.matrix.max.rows = 6)
library(readxl)

In [57]:
players_data <- read_csv("data/players.csv")
sessions_data <- read_csv("data/sessions.csv")

sessions_data

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


hashedEmail                                                     
1    bfce39c89d6549f2bb94d8064d3ce69dc3d7e72b38f431d8aa0c4bf95ccee6bf
2    36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
3    f8f5477f5a2e53616ae37421b1c660b971192bd8ff77e3398304c7ae42581fdc
⋮    ⋮                                                               
1533 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1534 fd6563a4e0f6f4273580e5fedbd8dda64990447aea5a33cbb5e894a3867ca44d
1535 36d9cbb4c6bc0c1a6911436d2da0d09ec625e43e6552f575d4acc9cf487c4686
     start_time       end_time         original_start_time original_end_time
1    30/06/2024 18:12 30/06/2024 18:24 1.71977e+12         1.71977e+12      
2    17/06/2024 23:33 17/06/2024 23:46 1.71867e+12         1.71867e+12      
3    25/07/2024 17:34 25/07/2024 17:57 1.72193e+12         1.72193e+12      
⋮    ⋮                ⋮                ⋮                   ⋮                
1533 28/07/2024 15:36 28/07/2024 15:57 1.72218e+12         1.72218e+12      
1534 25/07/2024 06:15 25/07/2024 06:22 1.72189e+12         1.72189e+12      
1535 20/05/2024 02:26 20/05/2024 02:45 1.71617e+12         1.71617e+12

## Data Description

In [56]:
players_data_mean <- players_data |>
    select(played_hours, Age) |>
    map_df(~mean(.x, na.rm = TRUE)) |>
    round(2)

players_data_experience_mean <- players_data |>
    select(experience, played_hours, Age) |>
    group_by(experience) |>
    summarise(across(c(played_hours, Age), mean, na.rm = TRUE)) |>
    mutate(across(where(is.numeric), ~ round(.x, 2)))

players_data_mean
players_data_experience_mean

played_hours Age  
1 5.85         21.14

experience played_hours Age  
1 Amateur     6.02        21.37
2 Beginner    1.25        21.66
3 Pro         2.60        16.92
4 Regular    18.21        22.03
5 Veteran     0.65        20.96

In the data set `players.csv`, there are 196 rows and 7 columns. 4 *character*  variables (`experience`, `hashedEmail`, `name`, `gender`), 2 *double-precision*  variables (`played_hours`, `Age`), and 1 *logical*  variable (`subscribe`). The average value of the 2 *double-precision*  variables `played_hours` and `Age` are 5.85 hours and 21.14 years old, respectively. After grouping players according to their `experience`, we can see that regular players played the most and have the highest average age, with an average of 18.21 hours played and 22.03 years old.

The data set has several *NA*  data points. So when performing any data analysis, it is important to use the `na.rm = TRUE` code to ignore these data points. Also, the data set contains some unnecessary data, like players' Email and their names. These are unique information about each player that would not contribute to creating any trends and relations to other variables. Other than that, changing the *character*  variables into *factor*  variables might also help with data wrangling and graphing. 